packages/client/src/comp/AddressInputs.tsx

```diff
@@ -0,0 +1,114 @@
+import React from "react"
+
+import { BaseInput } from "./util/Input"
+import styled from "styled-components"
+import { cssQuery } from "./util/cssQuery"
+
+const FlexBox = styled.div`
+  display: flex;
+  flex-direction: row;
+  justify-content: center;
+  flex-wrap: wrap;
+  align-items: center;
+  margin: 1em -0.5em;
+`
+
+const FlexGrow = styled.div`
+  margin: 0 0.5em;
+`
+
+const FlexGrowFull = styled.div`
+  margin: 0 0.5em;
+  flex: 0.5;
+  ${cssQuery.small} {
+    flex: 1;
+  }
+`
+
+const FlexGrowHalf = styled.div`
+  margin: 0 0.5em;
+  flex: 0.25;
+  ${cssQuery.small} {
+    flex: 0.5;
+  }
+`
+
+interface Props {
```

Comment:
We're now using a controlled form for Alloy.  See Base.tsx on master.  You can probably pass in a single `onChange`, `onBlur`, and `value` and `invalid` that accept an entire object rather than just a string